### PR TITLE
[`flake8-pyi`] Implement `PYI066`

### PR DESCRIPTION
**Pull request by tusharsadhwani**
## Summary

- Implements `Y066` from `flake8-pyi` as `PYI066`
- Fixes `PYI006` not being raised for `elif` clauses. This would have conflicted with PYI006's implementation, so decided to do it in the same PR.

## Test Plan

`cargo test` / `cargo insta review`

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_pyi/rules/bad_version_info_comparison.rs [hunk 1]
```diff
@@ -57,8 +58,59 @@ impl Violation for BadVersionInfoComparison {
...
+/// writing multiple versions of code while comparing Python versions.
+///
+/// ## Example
+///
+/// ```python
+/// import sys
+///
+/// if sys.version_info < (3, 10):
+///
+///     def read_data(x, *, preserve_order=True):
```
Comment (tusharsadhwani) [inline crates/ruff_linter/src/rules/flake8_pyi/rules/bad_version_info_comparison.rs:76]: I don't think this formatting is the best, but the `mkdocs` check is failing if I change this in any way.

Comment (AlexWaygood) [reply crates/ruff_linter/src/rules/flake8_pyi/rules/bad_version_info_comparison.rs:76]: I opened https://github.com/astral-sh/ruff/issues/11568 to track this; I've also bumped into this in the past :-)

Comment (tusharsadhwani) [reply crates/ruff_linter/src/rules/flake8_pyi/rules/bad_version_info_comparison.rs:76]: feel free to revert the last commit to format it correctly as needed.

**Comment by charliermarsh** [comment]
I can merge this one too.

**Pull request merged**